### ISSUE TITLE
Non-standard ID field names cause errors.

### ISSUE THREAD

**Issue by KyleBrost** [graphql-nexus/nexus-plugin-prisma#574]
I'm running a the following query on an apollo-server instance configured with prisma, nexus-prisma, and MySQL.

```
query {
  user(where : {user_id:"ck50s5bh70001ucvis0k7mesf"}) {
    user_id
    user_email
    user_customer {
      customer_id
    }
  }
}
```

ERROR
```
Invalid `new CustomerClient()` invocation in C:\working\alpha-api\node_modules\@prisma\photon\index.js:196:16

  192 const prefix = this._path.includes('select') ? 'select' : this._path.includes('include') ? 'include' : 'select';
  193 const path = [...this._path, prefix, 'user_customer'];
  194 const newArgs = runtime_1.deepSet(this._args, path, args || true);
  195 this._isList = false;
→ 196 return new CustomerClient(this._dmmf, this._fetcher, this._queryType, this._rootField, this._clientMethod, newArgs, path, this._isList);{
      where: {
        id: undefined,
    ?   user_id?: ID,
    ?   user_email?: String
      },
      select: {
        user_customer: {}
      }
    })

Argument where of type UserWhereUniqueInput needs at least one argument. Available args are listed in green.

Note: Lines with ? are optional.

```

I have attached the various files that could be important.  I'm confused about the error because I thought that nexus-prisma automatically resolved child objects.  Am I mistaken or missing something?

schema.prisma
```
datasource db {
  provider = "mysql"
  url      = "mysql://username:password@host1.example.com/prisma"
}

generator photon {
  provider = "photonjs"
}

model User {
  user_id    String  @default(cuid()) @id
  user_email String  @unique

  user_customer Customer?
  employee Employee?
}

model Employee {
  employee_id String @default(cuid()) @id
  user User?
}

model Customer {
  customer_id String @default(cuid()) @id
  customer_name String
  user User?
}
```

schema.ts
```
import { nexusPrismaPlugin } from 'nexus-prisma'
import { idArg, makeSchema, objectType, stringArg } from 'nexus'


export const User = objectType({
  name: 'User',
  definition(t) {
    t.model.user_id()
    t.model.user_email()
    t.model.user_customer()
    t.model.employee()
  },
})

export const Employee = objectType({
  name: 'Employee',
  definition(t) {
    t.model.employee_id()
    t.model.user()
  },
})

export const Customer = objectType({
  name: 'Customer',
  definition(t) {
    t.model.customer_id()
    t.model.customer_name()
    t.model.user()
  },
})  

const Query = objectType({
  name: 'Query',
  definition(t) {
    t.crud.user()
    t.crud.users()
  },
})

const Mutation = objectType({
  name: 'Mutation',
  definition(t) {
    t.crud.createOneUser({ alias: 'signupUser' })
  },
})

export const schema = makeSchema({
  types: [Query, Mutation, Customer, Employee, User],
  plugins: [nexusPrismaPlugin()],
  outputs: {
    schema: __dirname + '/generated/schema.graphql',
    typegen: __dirname + '/generated/nexus.ts',
  },
  typegenAutoConfig: {
    contextType: 'Context.Context',
    sources: [
      {
        source: '@prisma/photon',
        alias: 'photon',
      },
      {
        source: require.resolve('./context'),
        alias: 'Context',
      },
    ],
  },
})
```

**Comment by KyleBrost** [comment]
Doing more research and trying to figure this out, I found that when the @id field has a name other than "id" it does not resolve the nested object.

So when the id is 'customer_id' it fails.  When the id is 'id' it all works.

The prisma data-modeling says an ID field can be any name.  I think Nexus-Prisma is not mapping things correctly.

https://github.com/prisma/prisma2/blob/master/docs/data-modeling.md

**Comment by KyleBrost** [comment]
So there has been no insight on how to solve this issue.  What are the next steps?  Where in the code would I find where it gets the field name for the "id" from the prisma schema?  Any hints would be helpful, maybe I can fix it.

**Comment by KyleBrost** [comment]
I believe the issue is related to prisma2 and not directly nexus-prisma.  I have raised this at prisma with examples.

https://github.com/prisma/prisma2/issues/1355

**Comment by jasonkuhrt** [comment]
closed by #590
